test/TestProjects/MgmtMockTest/tests/Generated/Mock/ManagedHsmResourceMockTests.cs

```diff
@@ -27,54 +27,65 @@ public ManagedHsmResourceMockTests(bool isAsync) : base(isAsync, RecordedTestMod
+            var managedHsmResource = GetArmClient().GetManagedHsmResource(managedHsmResourceId);
+            await managedHsmResource.GetAsync();
         }
 
         [RecordedTest]
-        public async Task Update()
+        public async Task GetManagedHsms()
         {
-            // Example: Update an existing managed HSM Pool
-            var managedHsmResourceId = MgmtKeyvault.ManagedHsmResource.CreateResourceIdentifier("00000000-0000-0000-0000-000000000000", "hsm-group", "hsm1");
-            var managedHsmResource = GetArmClient().GetManagedHsmResource(managedHsmResourceId);
-            MgmtKeyvault.ManagedHsmData data = new MgmtKeyvault.ManagedHsmData(location: AzureLocation.WestUS)
-            {
-            };
-            data.Tags.ReplaceWith(new Dictionary<string, string>()
-            {
-                ["Dept"] = "hsm",
-                ["Environment"] = "dogfood",
-                ["Slice"] = "A",
-            });
+            // Example: List managed HSM Pools in a subscription
 
-            await managedHsmResource.UpdateAsync(WaitUntil.Completed, data);
+            var subscriptionResourceId = SubscriptionResource.CreateResourceIdentifier("00000000-0000-0000-0000-000000000000");
+            var subscriptionResource = GetArmClient().GetSubscriptionResource(subscriptionResourceId);
+            await foreach (var _ in subscriptionResource.GetManagedHsmsAsync())
+            {
+            }
         }
 
         [RecordedTest]
         public async Task GetMHSMPrivateLinkResourcesByMhsmResource()
         {
             // Example: KeyVaultListPrivateLinkResources
-            var managedHsmResourceId = MgmtKeyvault.ManagedHsmResource.CreateResourceIdentifier("00000000-0000-0000-0000-000000000000", "sample-group", "sample-mhsm");
-            var managedHsmResource = GetArmClient().GetManagedHsmResource(managedHsmResourceId);
 
+            var managedHsmResourceId = MgmtMockTest.ManagedHsmResource.CreateResourceIdentifier("00000000-0000-0000-0000-000000000000", "sample-group", "sample-mhsm");
+            var managedHsmResource = GetArmClient().GetManagedHsmResource(managedHsmResourceId);
             await foreach (var _ in managedHsmResource.GetMHSMPrivateLinkResourcesByMhsmResourceAsync())
             {
             }
         }
+
+        [RecordedTest]
+        public async Task Update()
+        {
+            // Example: Update an existing managed HSM Pool
+
+            var managedHsmResourceId = MgmtMockTest.ManagedHsmResource.CreateResourceIdentifier("00000000-0000-0000-0000-000000000000", "hsm-group", "hsm1");
+            var managedHsmResource = GetArmClient().GetManagedHsmResource(managedHsmResourceId);
+            await managedHsmResource.UpdateAsync(WaitUntil.Completed, new ManagedHsmData(new AzureLocation("placeholder"))
+            {
+                Tags =
+{
```

Comment:
Is this code style expected?

Reply:
Known issue: This is a known issue: https://github.com/Azure/autorest.csharp/issues/2377
We cannot fix this until Roslyn fixes this